content/blog/2018/10/2018-10-09-telemetry.adoc

```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+title: >
+  Jenkins telemetry for higher quality changes
+tags:
+- core
+author: daniel-beck
+---
+
+One of the major strengths of Jenkins is its customizability and extensibility.
+With its plugin ecosystem and long list of (possibly hidden) options, Jenkins can be used for a wide range of use cases.
+
+The downside of all this flexibility is that, not knowing how users use Jenkins, we mostly rely on issues filed in our bug tracker to know when things go wrong.
+And over the years, quite a few things went wrong.
+The worst of these are security fixes with unintended side effects.
+Unlike regular changes, it's not really feasible to roll back security fixes, so users sometimes had to choose between security and functionality.
+But even changes developed in the open, such as the introduction of link:/blog/2018/01/13/jep-200/[JEP-200], didn't go as smoothly as we hoped.
+link:/blog/2018/08/31/shifting-gears/[With big changes in the works], it's more important than ever for us to have a better idea how Jenkins is used to deliver major changes safely.
+
+link:/projects/evergreen/[Jenkins Evergreen] solves this in some ways by being always connected to the Jenkins project and reporting back telemetry (mostly errors, allowing us to quickly react).
+But that project is still pretty new, and its goal of being a more standardized Jenkins does not represent the breadth of configurations of the general user base.
+
+## Uplink telemetry
+
+So we recently extended the existing, very limited anonymous usage statistics by adding a simple, extensible telemetry reporting client.
+We're calling it _Uplink telemetry_, based on the name of https://github.com/jenkins-infra/uplink/[the service it reports its data to].
+It made its debut in Jenkins 2.143.
+
+Uplink telemetry is designed to collect data in _trials_:
+a well defined set of technical data with a specific purpose, and a start and end date of the collection.
+Detailed information explaining the scope and purpose of currently active trials is available in the inline help for the usage statistics control in the global configuration.
+
+image::/images/post-images/2018-10-09/telemetry.png[Screenshot of detailed Uplink telemetry trial description in the Jenkins inline help, role=center]
```

Comment:
Screen shot seems a bit large, also doesn't work too well with the flow of the text as its content is only described later in the post. Should be fine though?

Reply:
We can shrink it down a bit, right?